hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveMetastoreBasedLockProvider.java

```diff
@@ -192,6 +192,11 @@ private void acquireLockInternal(long time, TimeUnit unit, LockComponent lockCom
           throw e;
         }
       }
+    } finally {
+      // it is better to release WAITING lock, otherwise hive lock will hang forever
+      if (this.lock != null && this.lock.getState() != LockState.ACQUIRED) {
+        hiveClient.unlock(this.lock.getLockid());
```

Comment:
Does this hiveClient.unlock(id) throw an exception that the lock is not acquired ? In the test case below, are you generating this scenario where the LOCK is in WAITING stage and you try to release it